src/android/java/io/jxcore/node/BtToRequestSocket.java

```diff
@@ -67,75 +63,64 @@ public BtToRequestSocket(BluetoothSocket socket, BtSocketDisconnectedCallBack ha
         mmOutStream = tmpOut;
     }
 
-    public void setPort(int port) {
-        mHTTPPort = port;
-    }
-
     public void run() {
+
         try {
             print_debug("start Socket with port: " + mHTTPPort);
+            mHTTPPort = 0; // just to make sure its zero
             srvSocket = new ServerSocket(mHTTPPort);
             mHTTPPort = srvSocket.getLocalPort();
             print_debug("mHTTPPort  set to : " + mHTTPPort);
         } catch (Exception e) {
```

Comment:
:+1: 

I'm closing this comment because I opened a separate one where I call out the reasons why I believe that even the new Exception code is unfortunately not quite right.